Protos/V1/submitter_service.proto

```diff
@@ -149,4 +159,42 @@ message WaitRequest {
 	TaskFilter filter = 1;
 	bool stop_on_first_task_error = 2;
 	bool stop_on_first_task_cancellation = 3;
-}
\ No newline at end of file
+}
+
+/**
+ * Request to watch result states
+ * It contains the list of result ids you want to watch
+ *   and some options to filter out some events
+ */
+message WatchResultRequest {
+  message Options {
+    // List of statuses to ignore in the replies
+    repeated ResultStatus.ResultStatus ignore_statuses = 1;
+    // Skip the sending of result states at the beginning
+    optional bool skip_current_state = 2;
```

Comment:
In theory, I agree with you, but I can imagine a few use cases where this would be useful. For instance when a user creates results, starts to watch, and only then starts the tasks that will produce those results.